plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -1137,12 +1137,12 @@ public static String varcharLiteral(String value)
         return "'" + value.replace("'", "''") + "'";
     }
 
-    protected static Optional<String> escapeNamePattern(Optional<String> name, String escape)
+    protected Optional<String> escapeObjectNameForMetadataQuery(Optional<String> name, String escape)
     {
-        return name.map(string -> escapeNamePattern(string, escape));
+        return name.map(string -> escapeObjectNameForMetadataQuery(string, escape));
     }
 
-    private static String escapeNamePattern(String name, String escape)
+    protected String escapeObjectNameForMetadataQuery(String name, String escape)
```

Comment:
Extract an initial commit for `Rename escapeNamePattern to escapeObjectNameForMetadataQuery [and make it an extension point]` (part in brackets optional or move to commit body if too long).

Reply:
Nevermind, ignore this comment - doesn't make a lot of sense as a standalone commit.